src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,42 @@
 
 class Animal {
   // write your code here
+  health = 100;
+
+  constructor(name) {
+    // this = {}
+    // this.__proto__ = Animal.prototype
+    // this.health = 100
+    this.name = name;
+    Animal.alive.push(this);
+    // return this
+  }
+
+  static alive = [];
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name) {
```

Comment:
Maybe hidden better add as a parameter to constructor with default value false, to make code more understandable.